redash/models.py

```diff
@@ -757,6 +776,190 @@ def __unicode__(self):
         return unicode(self.id)
 
 
+class AccessPermission(BaseModel):
+
+    id = peewee.PrimaryKeyField()
+    object_type = peewee.CharField(index=True)
+    object_id = peewee.IntegerField(index=True)
+    object = GFKField('object_type', 'object_id')
+    access_type = peewee.CharField()
+    grantor = peewee.ForeignKeyField(User, related_name='grantor')
+    grantee = peewee.ForeignKeyField(User, related_name='grantee')
+
+    ACCESS_TYPE_VIEW = 'view'
+    ACCESS_TYPE_MODIFY = 'modify'
+    ACCESS_TYPE_DELETE = 'delete'
+
+    class Meta:
+        db_table = 'access_permissions'
+
+    @staticmethod
+    def grant_permission(object_type, object_id, access_type, grantee, grantor):
+        perm = AccessPermission()
+        perm.object_type = object_type
+        perm.object_id = object_id
+        perm.access_type = access_type
+        perm.grantor = grantor
+        perm.grantee = grantee
+        perm.save()
+
+    @staticmethod
```

Comment:
It should be a `@classmethod` and then use the `cls` function parameter instead of `AccessPermission`.